app/modules/intelligence/agents/custom_agents/custom_agents_service.py

```
@@ -141,22 +411,58 @@ async def update_agent(
+            logger.info(f"Converted agent schema visibility: {result.visibility}")
+            return result
         except SQLAlchemyError as e:
             self.db.rollback()
             logger.error(f"Database error while updating agent {agent_id}: {str(e)}")
             raise
+        except Exception as e:
+            self.db.rollback()
+            logger.error(f"Unexpected error while updating agent {agent_id}: {str(e)}")
+            raise HTTPException(
+                status_code=500, detail=f"Failed to update agent: {str(e)}"
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)